In [9]:
# Librerías necesarias para nuestro análisis:
import numpy as np
import pandas as pd
import os

# Modelos clasificatorios:
from sklearn.metrics import cohen_kappa_score
from sklearn.metrics import f1_score
from sklearn.metrics.pairwise import cosine_distances
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import classification_report
from sklearn.preprocessing import StandardScaler
from sklearn.pipeline import make_pipeline
from sklearn.tree import DecisionTreeClassifier
from sklearn.metrics import accuracy_score
from sklearn.metrics import confusion_matrix


COEFICIENTES PARA ESTUDIAR LA CALIDAD DE LOS DATOS DEL DATASET EN TÉRMINOS ANALIZAR Y PREDICIÓN 

In [5]:
# Cargar el dataset a analizar:
df_clean = pd.read_csv('clean_data.csv')

In [16]:
#Vamos a probar el coeficiente Kappa
from sklearn.preprocessing import StandardScaler
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import cohen_kappa_score
from sklearn.model_selection import train_test_split

x = df_clean[['Gender', 'Age', 'EDUC', 'SES', 'MMSE', 'CDR', 'eTIV']] 
y = df_clean['Group'] 

# Escalar los datos
scaler = StandardScaler()
x_scaled = scaler.fit_transform(x)

x_train, x_test, y_train, y_test = train_test_split(x_scaled, y, test_size=0.25)

# Aumentar el número de iteraciones
modelo = LogisticRegression(max_iter=10000)
modelo.fit(x_train, y_train)

y_pred = modelo.predict(x_test)

kappa = cohen_kappa_score(y_test, y_pred)

print(f"El coeficiente kappa es: {kappa:.4f}")

El coeficiente kappa es: 0.7835


COEFICIENTE KAPPA. Conclusión: 0.7835 demuestra un porcentaje aceptable, por lo que sugiere que los clasificadores están alineados en sus predicciones. Valores cercanos a 1 señalan mayor concordancia que la esperada por azar, pero estos resultados también pueden llegar a ser muy aleatorios.

In [24]:
# Distancia del coseno:

from sklearn.metrics.pairwise import cosine_distances

vector1 = df_clean.iloc[0, :-1].values
vector2 = df_clean.iloc[1, :-1].values

distancia = cosine_distances([vector1], [vector2])[0, 0]

print(f"Distancia del coseno: {distancia}")

Distancia del coseno: 1.913671443531939e-10


 El resultado obtenido por la distancia del coseno está en notación científica, por lo que el valor es bastante pequeño, siendo la distancia entre los vectores cercana y eso indica que los registros se parecen bastante entre sí

In [35]:
#  Calculamos la medida F1

from sklearn.preprocessing import StandardScaler
from sklearn.pipeline import make_pipeline

x = df_clean.iloc[:, :-1]
y = df_clean.iloc[:, -1]

x_train, x_test, y_train, y_test = train_test_split(x, y, test_size=0.25)

# Crear un pipeline que primero escala los datos y luego aplica la Regresión Logística. 
# También aumenta el número máximo de iteraciones permitidas en el modelo de Regresión Logística a 1000.
modelo = make_pipeline(StandardScaler(), LogisticRegression(max_iter=1000))
modelo.fit(x_train, y_train)

y_pred = modelo.predict(x_test)

# Generar classification report
report = classification_report(y_test, y_pred)

print(report)


              precision    recall  f1-score   support

           0       0.86      0.98      0.92        45
           1       0.86      1.00      0.92        36
           2       1.00      0.08      0.14        13

    accuracy                           0.86        94
   macro avg       0.91      0.68      0.66        94
weighted avg       0.88      0.86      0.81        94



MEDIDA F1: para evaluar el rendimiento del modelo. Con una precisión que ronda del 85% al 98%, indiga un porcentaje de predicciones correcto. La exhaustividad del recall es de hasta poder alcanzar del 98% para la clase 0 y 100% para la clase 1. Esto indica que el modelo identifica correctamente la mayoría de los casos positivos.